documentation/modules/ROOT/pages/02-architecture.adoc

```diff
@@ -1,104 +1,255 @@
+
+@startuml
+!theme cerulean-outline
+skinparam linetype polyline
+
+cloud "Event Mesh" {
+  component "Knative Broker" as Broker
+  queue "Kafka" as Kafka
+}
+
+folder "Micro services" {
+  component "Drivers Service" as DriversService
+  database "Drivers DB" as DriversDB
+  component "Invoicing Service" as InvoiceService
+  database "Invoicing DB" as InvoiceDB
+  component "Notification Service" as NotificationService
+}
+
+component "Legacy system" {
+  component "Legacy App" as Legacy
+  database "Legacy DB" as DB
+}
+
+Legacy -down-> Broker: Publish events
+Legacy .right.> DB : Update data
+Broker .right.> Kafka : Persist events
+DriversService .left.> Broker: Publish events
+Broker --> DriversService: Route events
+Broker --> InvoiceService: Route events
+Broker --> NotificationService: Route events
+DriversService ..> DriversDB: Gets info about drivers
+InvoiceService ..> InvoiceDB: Update Invoice records
+@enduml
+////
+
+The diagram illustrates the example flow of events between the applications, the Knative _Event Mesh_, and the datastores which persist settled state of the system.
 
 [TIP]
 ====
-This is a Tip
+Notice the applications aren't pulling the events from the queue.
+In fact they aren't aware of any.
+The _Event Mesh_ is the one controlling the flow, and retrying when needed.
+
+There are *no additional* libraries needed to consume events from _Event Mesh_.
+The _Event Mesh_ pushes the events as _CloudEvents_ encoded as _REST_ messages.
 ====
 
 [NOTE]
 ====
-This is a NOTE
-====
+The exponential backoff algorithm used by _Event Mesh_ is configurable.
+It uses the following formula to calculate the backoff period: `+backoffDelay * 2^<numberOfRetries>+`, where the backoff delay is a base number of seconds, and number of retries is automatically tracked by the _Event Mesh_.
 
-[WARNING]
-====
-This is a WARNING
+A dead letter sink can also be configured to send events in case they exceed the maximum retry number, which is also configurable. 
 ====
 
-[IMPORTANT]
-====
-This is IMPORTANT
-====
+=== _Work Ledger_ analogy
+
+A good way of thinking about the _Event Mesh_ and its persistent queue backend is the _Work Ledger_ analogy.
+Like in olden days, the clerk was keeping his to-do work in the _Work Ledger_ (e.g. a tray for paper forms).
+Then he was picking the next form, and processing it, making changes within the physical file cabinets.
+In case of rare and unexpected issues (e.g. Invoicing dept being unavailable), the clerk would just put the data back onto the _Work Ledger_ to be processed later.
```

Comment:
```suggestion
In case of rare and unexpected issues (e.g. invoicing dept being unavailable), the clerk would just put the data back onto the _Work Ledger_ to be processed later.
```